astro/src/content/quickstarts/quickstart-php-drupal-web.mdx

```diff
@@ -0,0 +1,752 @@
+
+```shell
+vendor/bin/drush generate block
+```
+
+Enter the following information:
+
+```
+Welcome to block generator!
+
+Module machine name:
+changebank
+
+Block admin label:
+User Login Logout Block
+
+Plugin ID [changebank_user_login_logout_block]:
+userloginlogoutlock
+
+Plugin class [UserloginlogoutlockBlock]:
+UserLoginLogoutBlock
+
+Block category [Custom]:
+Custom
+
+Make the block configurable? [No]:
+No
+
+Would you like to inject dependencies? [No]:
+No
+
+Create access callback? [No]:
+No
+
+The following directories and files have been created or updated:
+
+- /opt/drupal/web/modules/changebank/src/Plugin/Block/UserLoginLogoutBlock.php
+```
+
+To generate the form, run the following command.
+
+```shell
+vendor/bin/drush generate form
+```
+
+You will be presented with the form wizard. Fill it out as follows:
+
+```
+Welcome to form generator!
+
+Module machine name:
+changebank
+
+Module name [Changebank]:
+Changebank
+
+Class [ExampleForm]:
+ChangebankForm
+
+Would you like to create a route for this form? [Yes]:
+No
+
+The following directories and files have been created or updated:
+
+- /opt/drupal/web/modules/changebank/src/Form/ChangebankForm.php
+```
+
+Next, generate a controller with the following command.
+
+```shell
+vendor/bin/drush generate controller
+```
+
+You will be presented with the controller wizard. Fill it out as follows:
+
+```
+Welcome to controller generator!
+
+Module machine name:
+changebank
+
+Module name [Changebank]:
+Changebank
+
+Class [ChangebankController]:
+ChangebankController
+
+Would you like to inject dependencies? [No]:
+No
+
+Would you like to create a route for this controller? [Yes]:
+No
+
+The following directories and files have been created or updated:
+
+- /opt/drupal/web/modules/changebank/src/Controller/ChangebankController.php
+```
+
+Finally, generate a routing file with the following command.
+
+```shell
+vendor/bin/drush generate routing
+```
+
+You will be presented with the routing wizard. Fill it out as follows:
+
+```
+Module machine name:
+changebank
+
+Module name [Changebank]:
+Changebank
+
+The following directories and files have been created or updated:
+
+- /opt/drupal/web/modules/changebank/changebank.routing.yml
+```
+
+### Customizing The Module
+
+Now that you have all the files you need in your custom module, you can customize them.
+
+You will be editing the following files:
+
+- `ChangebankForm.php`
+- `ChangebankController.php`
+- `changebank.routing.yml`
+- `UserEmailBlock.php`
+- `UserLoginLogoutBlock.php`
+- `changebank.module`
+
+#### Customize The Changebank Form
+
+In your text editor, select the `ChangebankForm.php` file located at `your-application/web/modules/changebank/src/Form/ChangebankForm.php`. Clear the existing boilerplate contents and add the following code.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/web/modules/custom/changebank/src/Form/ChangebankForm.php"}
```

Comment:
I think we could add 
```php
declare(strict_types = 1);
```
to every PHP file

Reply:
added strict type checking